### DISCUSSION TITLE
openView does not exist on widget contribution

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9618] [Q&A] [answered]
Hello,

I am in the process of making a widget open automatically when Theia is started (I followed the example from "getting-started-contribution.ts" but when I try to use "this.openView" I am given the following error "openView does not exist on type widget contrubution". What could be the possible causes for this? I am not really sure how to go about fixing it. Thanks!

**Comment by vince-fugnitto** [comment, marked as answer]
@sboudreaux96 do you mind pushing your code to a public repository for the community to help?
It is difficult to identify the problem without the sources.

**Reply by sboudreaux96** [reply]
Sure - it is the account widget within the root folder of https://github.com/sboudreaux96/Theia_Custom_Widgets/tree/master/CustomWidgets

**Reply by vince-fugnitto** [reply]
Like the `getting-started` please try:

```ts
@postConstruct()
async initializeLayout(app: FrontendApplication): Promise<void>{
    await this.openView({activate: false});
}

async onStart(app: FrontendApplication): Promise<void>{
    this.openView({reveal: true});
}
```

**Reply by sboudreaux96** [reply]
I have tried that and it didnt work (openView does not exist)

**Reply by vince-fugnitto** [reply]
> I have tried that and it didnt work (openView does not exist)

The change works for me (no build errors), I made sure to clean the repo, and build.

- `git clean -ffdx`
- apply changes from https://github.com/eclipse-theia/theia/discussions/9618#discussioncomment-899022
- `yarn`

<img width="1398" alt="Screen Shot 2021-06-21 at 12 49 07 PM" src="https://user-images.githubusercontent.com/40359487/122798887-19351e80-d28f-11eb-9447-125a389664b1.png">

**Reply by sboudreaux96** [reply]
ok I see now. this works! thanks!
